### PR TITLE
Enable tests with `iree-llvm-link-embedded=true` tests in Bazel.

### PR DESCRIPTION
**Pull request by MaheshRavishankar**

### CODE REVIEW

File: iree/test/e2e/xla_ops/BUILD [hunk 1]
```diff
@@ -93,75 +93,74 @@ iree_check_single_backend_test_suite(
...
+        ],
+        include = ["*.mlir"],
+        exclude = [
+            "round.mlir",
+        ],
+    ),
+    compiler_flags = [
+        "-iree-input-type=mhlo",
+        "-iree-llvm-link-embedded=true",
+    ],
```
Comment (GMNGeoffrey) [inline iree/test/e2e/xla_ops/BUILD:160]:
```suggestion
    data = ["@llvm-project//ld.lld"],
    compiler_flags = [
        "-iree-input-type=mhlo",
        "-iree-llvm-link-embedded=true",
        "-iree-llvm-embedded-linker-path=$(location @llvm-project//ld.lld)",
    ],
```

Comment (MaheshRavishankar) [reply iree/test/e2e/xla_ops/BUILD:160]: Yeah, thanks. I forgot about this. Is it enough to add that to `iree-opt`, `iree-translate` and `iree-run-mlir`? Thats what I did the last time.

Comment (MaheshRavishankar) [reply iree/test/e2e/xla_ops/BUILD:160]: Actually I added the dependency on `gen-lld` in `llvm-project/lld`. That seems to be the portable version?

**Comment by hcindyl** [comment]
I don't know why I missed this PR before and how I bump into it now. :P 

Coincidentally I just worked on another similar PR on the runtime side. I actually added https://github.com/google/iree/pull/7044/commits/0fb9422a1d6fd104641a65a5775c67196f650b93 so it is wired in the genrule. That's the same mechanism we use in cmake. I think we need in both places since the test targets in this PR don't use `iree_bytecode_module`

**Comment by MaheshRavishankar** [comment]
> I don't know why I missed this PR before and how I bump into it now. :P
> 
> Coincidentally I just worked on another similar [PR](https://github.com/google/iree/pull/7044) on the runtime side. I actually added [0fb9422](https://github.com/google/iree/commit/0fb9422a1d6fd104641a65a5775c67196f650b93) so it is wired in the genrule. That's the same mechanism we use in cmake. I think we need in both places since the test targets in this PR don't use `iree_bytecode_module`

I actually dont know how to fix the bazel side here. Everything should be in place to enable this by default.

**Comment by hcindyl** [comment]
hmm... actually, if you can test locally by cherry-picking my commit https://github.com/google/iree/commit/0fb9422a1d6fd104641a65a5775c67196f650b93. It should work on this PR since internally `iree_check_test` actually calls `iree_bytecode_module` https://github.com/google/iree/blob/main/build_tools/bazel/iree_check_test.bzl#L45. So my fix should help you as well.

**Comment by hcindyl** [comment]
Please see the presubmit result of https://github.com/google/iree/pull/7068. I will take the issue over. :)

**Comment by MaheshRavishankar** [comment]
> Please see the presubmit result of #7068. I will take the issue over. :)

Thats awesome. Thanks Cindy! I saw the PR. Just a minor nit on that. Ill close this PR.

**Pull request closed without merging**